bin/deploy

```diff
@@ -1,42 +1,79 @@
 #/bin/sh
 
-args=("$@")
-
-case ${args[0]} in
-PELAGOS)
-    echo 'Setting Encore PublicPath to PELAGOS'
-    export publicpath="--publicpath /pelagos-symfony/build"
-    ;;
-NAS)
-    echo 'Setting Encore PublicPath to NAS'
-    export publicpath="--publicpath /build"
-    ;;
-*)
-    echo 'Please specify a Build Environment!'
-    echo 'Options: [PELAGOS, NAS]'
-    exit;
-    ;;
-esac
-
-# Stop RabbitMQ consumers
-bin/console rabbitmq-supervisor:control stop --wait-for-supervisord
-
-# Run composer install to make sure all non-dev dependencies are installed and parameters are defined
-composer install --no-dev --optimize-autoloader
-
-# Handle Yarn
-yarn install
-yarn run encore prod $publicpath
-
-# Dump ENV into compiled format
-composer dump-env prod
-
-# Warn of any pending migrations, but don't apply them.
-bin/console doctrine:migrations:status
-
-# Rebuild Supervisor config for RabbitMQ consumers and start them
-bin/console rabbitmq-supervisor:rebuild --wait-for-supervisord
-
-# Rebuild elasticsearch indexes
-bin/console fos:elastica:reset
-bin/console fos:elastica:populate
+function deploy {
+    # Stop RabbitMQ consumers
+    bin/console rabbitmq-supervisor:control stop --wait-for-supervisord
+
+    # Run composer install to make sure all non-dev dependencies are installed and parameters are defined
+    composer install --no-dev --optimize-autoloader
+
+    # Handle Yarn
+    yarn install
+    yarn run encore prod $publicpath
+
+    # Dump ENV into compiled format
+    composer dump-env prod
+
+    # Warn of any pending migrations, but don't apply them.
+    bin/console doctrine:migrations:status
+
+    # Rebuild Supervisor config for RabbitMQ consumers and start them
+    bin/console rabbitmq-supervisor:rebuild --wait-for-supervisord
+
+    # Rebuild elasticsearch indexes
+    bin/console fos:elastica:reset
+    bin/console fos:elastica:populate
+}
+
+function checkdiff {
+    diff=`git diff`
+    if [ ! -z "$diff" ]
+    then
+        echo "Local changes found in $PWD. Not deploying."
+        exit 1
+    fi
+}
+
+if [ $# -eq 0 ]
+  then
+    echo "Please specify branch."
+    exit 1
+fi
+
+branch=$1
+
+# Save current directory before setting.
+export prevDir=$PWD
+
+# check for local changes
+cd /opt/pelagos
+checkdiff
+cd /opt/grp
+checkdiff
+
+# PELAGOS
+cd /opt/pelagos
+git fetch
+gitresult=`git checkout $branch 2>&1`
```

Comment:
Can we make the Git Tag to be a argument that is being passed into the command? just like PELAGOS/NAS